File: VB/WinSolution.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating

Namespace WinSolution.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub

		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			For i As Integer = 0 To 4
				Dim o As Order = ObjectSpace.CreateObject(Of Order)()
				o.Name = String.Format("Order{0}", i)
				If i Mod 2 = 0 Then
					o.SimpleBusinessAction()
				End If
			Next i
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module/Updater.cs
using System;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            using (UnitOfWork uow = new UnitOfWork(Session.DataLayer)) {
                for (int i = 0; i < 5; i++) {
                    Order o = new Order(uow);
                    o.Name = string.Format("Order{0}", i);
                    if (i % 2 == 0){
                        o.SimpleBusinessAction();
                    }
                }
                uow.CommitChanges();
            }
        }
    }
}
